tests/linker/mac/link all/LinkAllTest.cs

```diff
@@ -25,5 +29,41 @@ public void Calendars ()
 			Check ("HijriCalendar", false);
 			Check ("ThaiBuddhistCalendar", false);
 		}
+
+		[Test]
+		public void EnsureUIThreadException ()
+		{
+			// works on main/ui thread
+			NSApplication.EnsureUIThread ();
+
+			ThreadPool.QueueUserWorkItem ((v) => Tester.Test ());
+			Assert.IsTrue (Tester.mre.WaitOne (TimeSpan.FromSeconds (10)), "Successful wait");
+#if DEBUG
+			var expected_ex_thrown = true;
+#else
+			var expected_ex_thrown = false;
+#endif
```

Comment:
Approving, but @rolfbjarne could we have a comment of why is this added? I'm sure we will forget

Reply:
It's because the UI thread check exception only occur in debug builds; in release builds the linker removes the check.